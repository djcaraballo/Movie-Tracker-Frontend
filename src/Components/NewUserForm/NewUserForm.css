h2, h3, label {
  color: white;
}

.form-wrapper {
  height: 100vh;
  background: #000;
}

.form-input {
  height: 1.5em;
  margin-bottom: 1em;
  /*font-size: 1.5em;*/
  font-family: 'Lato', sans-serif;
}

.new-user-form {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 350px;
  justify-content: space-around;
  padding-bottom: 50px;
  flex-direction: column;
  width: 75%;
  margin: 4rem auto;
  background: #000;
  font-family: 'Lato', sans-serif;
}

.user-name-input,
.user-email-input,
.user-password-input,
.signup-button {
  border-radius: 14px;
  font-size: 1.5em;
  height: 40px;
  padding-bottom: 10px;
  padding-left: 20px;
  padding-top: 10px;
  width: 30rem;
}

.user-name-input {
  margin-top: 30px;
}

.signup-button {
  background-color: #beb47e;
  font-weight: bold;
  height: 58px;
  margin-bottom: 30px;
  width: 31.75rem;
}

@media (max-width: 600px) {
  .user-name-input,
  .user-email-input,
  .user-password-input {
    height: 30px;
    padding-bottom: 5px;
    padding-left: 10px;
    padding-top: 5px;
    width: 85%;
  }

  .signup-button {
    font-size: 1em;
    height: 48px;
    width: 86.75%;
  }

  .user-login-form {
    margin: 3rem auto;
    width: 85%;
  }

  .form-input {
    font-size: 1em;
  }
}

@media (max-width: 475px) {
  .new-user-form {
    margin: 2rem auto;
  }
}

@media (max-width: 350px) {
  .new-user-form {
    margin: 1rem auto;
  }
}

