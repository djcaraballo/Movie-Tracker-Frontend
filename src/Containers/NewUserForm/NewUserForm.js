import React, { Component } from 'react';

class NewUserForm extends Component {
  constructor(props) {
    super(props)
    this.state ={
      name: '',
      email: '',
      password: '',
    }
  }

  const handleSubmit = (e) => {
    const { name, value } = e.target
    const { handleSubmit } = this.props
    this.setState({ [name]: value })
  }

  render() {
    const { name, email, password } = this.state
    return (
      <form onSubmit={this.handleSubmit}>
        <input 
          name="name" 
          className="user-name-input" 
          value={name}
          onChange={this.handleChange} />
        <input 
          name="email" 
          className="user-email-input" 
          value={email}
          onChange={this.handleChange} />
        <input 
          name="password" 
          className="user-password-input" 
          value={password}
          onChange={this.handleChange} />
        <button>Submit</button>
      </form>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  handleSubmit: (name, email, password) => ({
    dispatch(newUser(name, email, password))
  })
}

export default connect (null, mapDispatchToProps)(NewUserForm)