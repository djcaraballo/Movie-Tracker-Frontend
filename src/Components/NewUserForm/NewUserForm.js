import React, {Component} from 'react';
import * as API from '../../Utils/API';
import './NewUserForm.css';

class NewUserForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: '',
      password: '',
      favorites: [],
      hasError: false,
    };
  }

  handleChange = e => {
    const {name, value} = e.target;
    this.setState({[name]: value});
  };

  handleSubmit = async e => {
    e.preventDefault();
    const result = await API.postNewUser(this.state);
    if (result.error) {
      this.setState({hasError: true});
    } else {
    }
  };

  render() {
    const {name, email, password, hasError} = this.state;

    return (
       <div className="form-wrapper">
         <form className="new-user-form" onSubmit={this.handleSubmit}>
          <input
            name="name"
            className="form-input user-name-input"
            value={name}
            onChange={this.handleChange}
            placeholder="Enter Full Name"
            required
          />
          {hasError ? <label htmlFor="Email">Email Already Taken</label> : null}
          <input
            name="email"
            id="Email"
            className="form-input user-email-input"
            value={email}
            onChange={this.handleChange}
            placeholder="Enter Email Address"
            required
          />
          <input
            name="password"
            type="password"
            className="form-input user-password-input"
            value={password}
            onChange={this.handleChange}
            placeholder="Enter Password"
          />
          <button className="signup-button">Sign Up!</button>
        </form>
      </div>
    );
  }
}

export default NewUserForm;
