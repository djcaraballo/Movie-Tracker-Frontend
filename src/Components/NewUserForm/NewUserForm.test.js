import React from 'react';
import ReactDOM from 'react-dom';
import NewUserForm from './NewUserForm';
import { shallow } from 'enzyme';

describe('NewUserForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NewUserForm />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});