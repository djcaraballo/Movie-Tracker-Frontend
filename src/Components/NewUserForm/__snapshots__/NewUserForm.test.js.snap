// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUserForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewUserForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="new-user-form"
        onSubmit={[Function]}
      >
        <input
          className="form-input user-name-input"
          name="name"
          onChange={[Function]}
          placeholder="Enter Full Name"
          required={true}
          value=""
        />
        <input
          className="form-input user-email-input"
          id="Email"
          name="email"
          onChange={[Function]}
          placeholder="Enter Email Address"
          required={true}
          value=""
        />
        <input
          className="form-input user-password-input"
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
        <button
          className="signup-button"
        >
          Sign Up!
        </button>
      </form>,
      "className": "form-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-input user-name-input"
            name="name"
            onChange={[Function]}
            placeholder="Enter Full Name"
            required={true}
            value=""
          />,
          null,
          <input
            className="form-input user-email-input"
            id="Email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email Address"
            required={true}
            value=""
          />,
          <input
            className="form-input user-password-input"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
            value=""
          />,
          <button
            className="signup-button"
          >
            Sign Up!
          </button>,
        ],
        "className": "new-user-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-name-input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Enter Full Name",
            "required": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-email-input",
            "id": "Email",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Email Address",
            "required": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up!",
            "className": "signup-button",
          },
          "ref": null,
          "rendered": "Sign Up!",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="new-user-form"
          onSubmit={[Function]}
        >
          <input
            className="form-input user-name-input"
            name="name"
            onChange={[Function]}
            placeholder="Enter Full Name"
            required={true}
            value=""
          />
          <input
            className="form-input user-email-input"
            id="Email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email Address"
            required={true}
            value=""
          />
          <input
            className="form-input user-password-input"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
            value=""
          />
          <button
            className="signup-button"
          >
            Sign Up!
          </button>
        </form>,
        "className": "form-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-input user-name-input"
              name="name"
              onChange={[Function]}
              placeholder="Enter Full Name"
              required={true}
              value=""
            />,
            null,
            <input
              className="form-input user-email-input"
              id="Email"
              name="email"
              onChange={[Function]}
              placeholder="Enter Email Address"
              required={true}
              value=""
            />,
            <input
              className="form-input user-password-input"
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              type="password"
              value=""
            />,
            <button
              className="signup-button"
            >
              Sign Up!
            </button>,
          ],
          "className": "new-user-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-name-input",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Enter Full Name",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-email-input",
              "id": "Email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter Email Address",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-password-input",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up!",
              "className": "signup-button",
            },
            "ref": null,
            "rendered": "Sign Up!",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
